src/Microsoft.VisualStudio.Threading.Tests/ReentrantSemaphoreTestBase.cs

```diff
@@ -227,23 +228,24 @@ public void ExitInAnyOrder_ExitInAcquisitionOrder()
         this.semaphore = this.CreateSemaphore(ReentrantSemaphore.ReentrancyMode.Freeform);
         this.ExecuteOnDispatcher(async delegate
         {
-            var releaser1 = new AsyncManualResetEvent();
+            var outerReleaser = new AsyncManualResetEvent();
             var innerReleaser = new AsyncManualResetEvent();
             Task innerOperation = null;
-            await this.semaphore.ExecuteAsync(delegate
+            await this.semaphore.ExecuteAsync(async delegate
             {
-                innerOperation = EnterAndUseSemaphoreAsync(innerReleaser);
-                return TplExtensions.CompletedTask;
+                innerOperation = EnterAndUseSemaphoreAsync(innerReleaser, outerReleaser);
+                await outerReleaser;
             });
             innerReleaser.Set();
             await innerOperation;
             Assert.Equal(1, this.semaphore.CurrentCount);
         });
 
-        async Task EnterAndUseSemaphoreAsync(AsyncManualResetEvent releaseEvent)
+        async Task EnterAndUseSemaphoreAsync(AsyncManualResetEvent releaseEvent, AsyncManualResetEvent signal)
```

Comment:
>signal [](start = 103, length = 6)

Given we take two of these, can we name this more distinctively? Perhaps `enteredEvent`?

Reply:
Also, I'd rather reorder these parameters so that they are used in-order.

---
In reply to: [201717015](https://github.com/Microsoft/vs-threading/pull/313#discussion_r201717015) [](ancestors = 201717015)